pages/api/studentPortal/quizzes/FirstLang.ts

```diff
@@ -0,0 +1,67 @@
+type QuizData = {
+    name?: string;
+    title?: string;
+    body?: string;
+    result?: string;
+    weight?: number;
+  };
+
+export const quizQuestions: QuizData[][] = [
+    [
+      {
+        title: "What Coding Language Should I Learn First?",
+        body: "Take this free quiz to find out what coding language you should learn first.",
+      },
+    ],
+    [
+      {
+        title: "Why do you want to learn coding?",
+        body: "Select all that apply.",
+      },
+      { name: "Learn an in-demand skill" },
+      { name: "Work in tech" },
+      { name: "Build programs for fun" },
+      { name: "I run a startup" },
+      { name: "Become a professional developer" },
+      { name: "Communicate better with technical coworkers" },
+      { name: "Start a side hustle" },
+    ],
+    [
+      {
+        title: "What field of work are you interested in?",
+        body: "Select 1-3 choices.",
+      },
+      { name: "Front end development", result: "JavaScript", weight: 1 },
+      { name: "Back end development", result: "JavaScript", weight: 1 },
+      { name: "Game development", result: "JavaScript", weight: 1 },
+      { name: "Mobile development", result: "JavaScript", weight: 1 },
+      { name: "Data analytics", result: "Python", weight: 1 },
+      { name: "Product management", result: "JavaScript", weight: 1 },
+      { name: "Digital marketing", result: "JavaScript", weight: 1 },
+      { name: "UX/UI design", result: "JavaScript", weight: 1 },
+      { name: "I don't know / Not sure yet", result: "JavaScript", weight: 1 },
+    ],
+    [
+      {
+        title: "What are you interested in building?",
+        body: "Select all that apply.",
+      },
+      { name: "Websites", result: "JavaScript", weight: 1 },
+      { name: "Storefront", result: "HTML/CSS", weight: 1 },
+      { name: "Mobile apps", result: "JavaScript", weight: 1 },
+      { name: "Games", result: "JavaScript", weight: 1 },
+      { name: "Email", result: "HTML/CSS", weight: 1 },
+      { name: "Tools to automate your life", result: "JavaScript", weight: 1 },
+      { name: "I don't know / Not sure yet", result: "JavaScript", weight: 1 },
+    ],
+  ];
+
+  export const selectionsPerPage = quizQuestions.map((pages) => {
+    return pages
+      .map((selections) => {
+        if (selections.name !== undefined && selections.name !== null) {
```

Comment:
this is not a clean way of getting selections. you have map through your array and look for all the objects that have a name, and you are assuming that is what you're looking for. your code is very prone to causing bugs or errors